internal/controllers/admin.go

```diff
@@ -0,0 +1,215 @@
+
+}
+
+func (s Server) GetAllActiveQuotas(ctx echo.Context) error {
+	resource := ctx.QueryParam("resource")
+	resourcefilter := ""
+	if resource != "" {
+		resourcefilter = ` and resource_types.name = '` + resource + `'`
+	}
+	username := ctx.QueryParam("username")
+	usernamefilter := ""
+	if username != "" {
+		usernamefilter = ` and users.username = '` + username + `'`
+	}
+	now := time.Now().Format("2006-01-02")
+
+	plandata := []AdminQuotaDetails{}
+
+	err := s.GORMDB.Debug().Raw(`select users.username as user_name, users.id as user_id, plans.name as plan_name, quotas.quota, resource_types.name as resource_name, resource_types.unit from user_plans
+	join plans on plans.id = user_plans.plan_id	
+	join quotas on user_plans.id=quotas.user_plan_id
+	join resource_types on resource_types.id=quotas.resource_type_id
+	join users on users.id = user_plans.user_id
+	where
+	user_plans.effective_start_date <=? and
+	user_plans.effective_end_date >=? `+usernamefilter+resourcefilter, now+"T00:00:00", now+"T23:59:59").Scan(&plandata).Error
+	if err != nil {
+		return ctx.JSON(http.StatusInternalServerError, model.ErrorResponse(err.Error(), http.StatusInternalServerError))
+	}
+	return ctx.JSON(http.StatusOK, model.SuccessResponse(plandata, http.StatusOK))
+}
+
+func (s Server) GetAllActiveUsage(ctx echo.Context) error {
+	resource := ctx.QueryParam("resource")
+	resourcefilter := ""
+	if resource != "" {
+		resourcefilter = ` and resource_types.name = '` + resource + `'`
+	}
+	username := ctx.QueryParam("username")
+
+	usernamefilter := ""
+	if username != "" {
+		usernamefilter = ` and users.username = '` + username + `'`
+	}
+	now := time.Now().Format("2006-01-02")
+
+	plandata := []AdminUsageDetails{}
+
+	err := s.GORMDB.Debug().Raw(`select users.username as user_name, users.id as user_id, plans.name as plan_name, usages.usage, resource_types.unit, resource_types.name as resource_name from user_plans 
+		join plans on plans.id = user_plans.plan_id 
+		join usages on user_plans.id=usages.user_plan_id 
+		join quotas on user_plans.id=quotas.user_plan_id 
+		join resource_types on resource_types.id=quotas.resource_type_id
+		join users on users.id = user_plans.user_id
+		where 
+		user_plans.effective_start_date <=? and 
+		user_plans.effective_end_date >=? `+usernamefilter+resourcefilter, now+"T00:00:00", now+"T23:59:59").Scan(&plandata).Error
+	if err != nil {
+		return ctx.JSON(http.StatusInternalServerError, model.ErrorResponse(err.Error(), http.StatusInternalServerError))
+	}
+	return ctx.JSON(http.StatusOK, model.SuccessResponse(plandata, http.StatusOK))
+
+}
+func (s Server) GetAllUserActivePlans(ctx echo.Context) error {
+	username := ctx.Param("username")
+	if username == "" {
+		return ctx.JSON(http.StatusBadRequest, model.ErrorResponse("Invalid Usernaem", http.StatusBadRequest))
+	}
+	now := time.Now().Format("2006-01-02")
+
+	plandata := []PlanDetails{}
+	err := s.GORMDB.Raw(`select plans.name,usage.usage,quotas.quota,resource_types.unit from
```

Comment:
Looks like there's a typo here:

```suggestion
	err := s.GORMDB.Raw(`select plans.name,usages.usage,quotas.quota,resource_types.unit from
```